aten/src/ATen/native/nested/cuda/NestedTensorTransformerFunctions.cpp

```diff
@@ -101,5 +102,106 @@ Tensor nested_from_padded_cuda(
     return at::native::nested_from_padded_generic(padded, sizes);
   }
 }
+
+Tensor _collapse_two_dims_3(const Tensor& input, int64_t dim1, int64_t dim2) {
+  TORCH_CHECK(dim1 > 0, "dim1: Cannot collapse dim 0.");
+  TORCH_CHECK(dim2 > 0, "dim2: Cannot collapse dim 0.");
+  TORCH_CHECK(dim2 - 1 == dim1, "dim2 must be one more than dim1.")
+  TORCH_CHECK(dim1 == 1, "dim1 must be 1.")
+  TORCH_CHECK(input.dim() == 3, "Expected input to be 3 dim.");
+
+  auto* nt_input = get_nested_tensor_impl(input);
+  TORCH_CHECK(nested_tensor_impl_is_contiguous(nt_input));
+  Tensor nt_sizes = nt_input->get_nested_size_tensor();
+
+  Tensor sizes_dim1 = at::native::narrow(nt_sizes, 1, 0, 1);
+  Tensor sizes_dim2 = at::native::narrow(nt_sizes, 1, 1, 1);
+
+  Tensor new_nt_sizes;
+  if (dim1 == 1) {
+    Tensor collapsed_sizes = sizes_dim1 * sizes_dim2;
+    new_nt_sizes = collapsed_sizes.contiguous();
```

Comment:
why wouldn't it be contiguous already? it's a newly-allocated Tensor

Reply:
Good catch, this is a relic from pytorch/nestedtensor